libkbfs/block_journal.go

```diff
@@ -72,18 +59,20 @@ type blockJournal struct {
 	log      logger.Logger
 	deferLog logger.Logger
 
-	j    diskJournal
-	refs map[BlockID]blockRefMap
+	// j is the main journal.
+	j diskJournal
 
+	// saveUntilMDFlush, when non-nil, prevents garbage collection
+	// of blocks. When removed, all the referenced blocks are
+	// garbage-collected.
+	//
+	// TODO: We only really need to save a list of IDs, and not a
```

Comment:
Can you do it separately, since this PR is already pretty big?

Re. "full journal", since it appeared that the list of block IDs was static, that's what led me down that train of thought. But if you decide to fix the bug by appending to both journals, then that would indicate that the current solution is fine, or perhaps change it slightly by storing not a blockOp in the second disk journal, but a struct with a list of block IDs or something.

Otherwise, then we may want to eventually write out a single file or something, but it's not really a high priority; just probably needs a comment.